### PR TITLE
Replace usage of gpr_cmdline with gflags, and remove gpr_cmdline from…

### PR DESCRIPTION
**Pull request by yashykt**
Replace usage of gpr_cmdline with gflags, and remove gpr_cmdline from grpc

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: test/core/bad_ssl/server_common.cc [hunk 1]
```diff
@@ -28,19 +29,23 @@
  * configured to cause some failure case in the SSL connection path.
  */
 
+// In some distros, gflags is in the namespace google, and in some others,
+// in gflags. This hack is enabling us to find both.
+namespace google {}
```
Comment (ncteisen) [inline test/core/bad_ssl/server_common.cc]: Rather than duplicate this in all of the files, use a common headers that has this comment and namespace hack

Comment (ncteisen) [reply test/core/bad_ssl/server_common.cc]: Would suggest adding to test/core/util/test_config.h actually. Just like is done in test/cpp/util/test_config.h

File: test/core/network_benchmarks/low_level_ping_pong.cc [hunk 2]
```diff
@@ -44,6 +45,37 @@
...
 #include "test/core/util/histogram.h"
 
+// In some distros, gflags is in the namespace google, and in some others,
+// in gflags. This hack is enabling us to find both.
+namespace google {}
+namespace gflags {}
+using namespace google;
+using namespace gflags;
+
+static const char* read_strategy_usage =
```
Comment (ncteisen) [inline test/core/network_benchmarks/low_level_ping_pong.cc]: Why was this code above the fd_pair struct? Seems unrelated

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by yashykt** [comment]
Update: Before these changes can be made, we would want to finish the move to foundry since the current tests use the language field in build.yaml to check if they are core tests or C++ tests. To use gflags, we would need the language field to be C++, which would mean that we would not be able to distinguish between core tests and C++ tests.

**Comment by vjpai** [comment]
Is the plan to eventually revive this PR or just restart from scratch by, for example, moving increasingly to gtest?

**Comment by yashykt** [comment]
I think the PR was in a state where it could be revived. The reason it's not been revived yet is mainly because some people (me included) use run_tests.py with the option -lc to run core tests. Doing this would change that.

**Comment by vjpai** [comment]
Is this PR still alive?

**Comment by vjpai** [comment]
Removing reviewers until it's ready to use
